orttraining/orttraining/training_ops/cuda/tensor/gather_grad.cc

```diff
@@ -27,14 +27,65 @@ ONNX_OPERATOR_KERNEL_EX(
     kCudaExecutionProvider,
     (*KernelDefBuilder::Create())
         .InputMemoryType(OrtMemTypeCPUInput, 0)
+        .InputMemoryType(OrtMemTypeCPU, 3)
```

Comment:
Let's add some comments here, 
specifically mentioning the difference between OrtMemTypeCPUInput and OrtMemTypeCPU.